hack/ci/windows.ps1

```diff
@@ -616,6 +638,15 @@ Try {
     Write-Host -ForegroundColor Green "INFO: Args: $dutArgs"
     New-Item -ItemType Directory $env:TEMP\daemon -ErrorAction SilentlyContinue  | Out-Null
 
+    # Start containerd first
+    if (-not ("$env:DOCKER_WINDOWS_CONTAINERD_RUNTIME" -eq "")) {
+        Start-Process "$env:TEMP\binary\containerd.exe" `
```

Comment:
Do we need to make sure `containerd-shim-runhcs-v1.exe` is in `PATH`, or is `containerd.exe` smart enough to look next to itself for that? (So we don't accidentally get one from the host.)

Reply:
https://github.com/containerd/containerd/blob/v1.5.5/runtime/v2/shim/util.go#L66-L100 uses `os/exec.LookPath` first, and _then_ if not found, it checks next to the containerd binary. So yes, I guess it would find a system-`$PATH`-installed containerd shim before the one next to containerd.exe.

Reply:
Good catch. `$env:PATH` gets already overwritten on https://github.com/moby/moby/blob/ba2adeebdb8d4ec442f1d3107405ae55d0c1a0c2/hack/ci/windows.ps1#L570
and
https://github.com/moby/moby/blob/ba2adeebdb8d4ec442f1d3107405ae55d0c1a0c2/hack/ci/windows.ps1#L843
so combining those together and move it to be done before containerd.exe is started should be enough.

Will verify that and update PR.

Reply:
Fixed on latest commit.

Reply:
@tianon is that good now?

Reply:
> https://github.com/containerd/containerd/blob/v1.5.5/runtime/v2/shim/util.go#L66-L100 uses os/exec.LookPath first, and then if not found, it checks next to the containerd binary. So yes, I guess it would find a system-$PATH-installed containerd shim before the one next to containerd.exe.

Hm.. reminds me of a security fix in Go 1.15. Opened https://github.com/containerd/containerd/pull/5906 to fix that 😅